test/create_db.sh

```diff
@@ -3,6 +3,37 @@ set -o errexit
 cd $(dirname $0)/..
 source test/db-common.sh
 
+# posix compliant escape sequence
+esc=$'\033'"["
+
+function print_heading() {
+  echo
+  echo -e "${esc}0;34;1m"$1"${esc}0m"
+}
+
+function get_migrations() {
+  migrations=( $(find "$dbpath"/migrations -mindepth 1 -maxdepth 1 -not -path '*/\.*'  2> /dev/null | sort) )
+}
+
+function create_empty_db() {
+  create_script="drop database if exists \`${db}\`; create database if not exists \`${db}\`;"
+  mysql $dbconn -e "$create_script" || die "unable to create ${db}"
+  echo "Created empty "$db" database"
+}
+
+function get_container_version() {
+  container_version="$(goose -path="$dbpath" -env="$dbenv" dbversion | sed 's/goose: dbversion //')"
+}
+
+function apply_migrations() {
+  if [ ! -z "${migrations[@]+x}" ]
```

Comment:
I don't know what `+x` does here, and it's hard to look up. I'm curious to know what it does! But also given the property that it's hard to look up, is there another clearer way we can do it?

Also `! -z` is equivalent to `-n`:

```
      -z STRING      True if string is empty.
    
      -n STRING
         STRING      True if string is not empty.
```

Reply:
`+x` is a trick in older versions of bash (though honestly still in version 4.4). https://git.savannah.gnu.org/cgit/bash.git/tree/CHANGES?id=3ba697465bc74fab513a26dea700cc82e9f4724e#n878

the only truly safe idiom for getting around this is `${arr[@]+"${arr[@]}"}` but `${arr[@]+x}` is broadly compatible and works fine.

Doesn't much matter though going forward, I replaced this code with the newer test:
```
if [[ "migrations[@]" ]]
then
  action
fi
```